lib/rspec/mocks/proxy.rb

```diff
@@ -371,20 +371,34 @@ def original_method_handle_for(message)
 
         return super unless unbound_method
         unbound_method.bind(object)
+      rescue TypeError
+        if RUBY_VERSION == '1.8.7'
+          # In MRI 1.8.7, a singleton method on a class cannot be rebound to its subclass
+          if unbound_method && unbound_method.owner.ancestors.first != unbound_method.owner
+            # This is a singleton method; we can't do anything with it
+            # But we can work around this using a different implementation
+            double = method_double_from_ancestor_for(message)
+            return object.method(double.method_stasher.send(:stashed_method_name))
```

Comment:
The tricky thing about adding a new method to the method double is that `stashed_method_name` only exists when you are running on Ruby 1.8! So do we conditionally define another method on `MethodDouble` which also exists only on 1.8? Seems yucky.